### PR TITLE
Additional changes to Span<T> and Span<T>.Enumerator

### PR DESCRIPTION
**Pull request by rpetrusha**
## Additional changes to Span<T> and Span<T>.Enumerator

@BillWagner @ahsonkhan @stephentoub

### CODE REVIEW

File: xml/System/Span`1.xml [hunk 1]
```diff
@@ -858,7 +870,7 @@ For a `Span<Char>`, the `ToString` method returns a <xref:System.String> that co
         <remarks>
           <format type="text/markdown"><![CDATA[  
 
-If the source and `destination` overlap, this method places the original values in a temporary location before `destination` is overwritten.
+If the source and `destination` overlap, the source is written to `destination` without any loss of data.
```
Comment (ahsonkhan) [inline xml/System/Span`1.xml]: I am not sure if `without any loss of data` is clear. The contents of the destination to get overwritten by whatever was in the source and hence are lost. Thoughts?

File: xml/System/Span`1+Enumerator.xml [hunk 2]
```diff
@@ -29,19 +29,25 @@ The C# [foreach](~/docs/csharp/language-reference/keywords/foreach-in.md) of the
...
 
-The enumerator does not have exclusive access to the <xref:System.Span%601>; therefore, enumerating through a span is intrinsically not a thread-safe procedure. To guarantee thread safety during enumeration, you can lock the span during the entire enumeration. To allow the span to be accessed by multiple threads for reading and writing, you must implement your own synchronization. 
+The enumerator does not have exclusive access to the <xref:System.Span%601>. In addition, the underlying data on which the span is based can also be modified. Therefore, enumerating through a span is intrinsically not a thread-safe procedure. To guarantee thread safety during enumeration, you must implement your own synchronization. For example, the following code has a race condition. It does not ensure that the span will be enumerated before the `ClearContents` method executes. As a result,   
 
-Unlike some other enumerator structures in .NET, <xref:System.Span%601.Enumerator>:
+[!code-csharp[System.Span<T>.Enumerator-no-synchronization](~/samples/snippets/csharp/api/system/span.enumerator/nosync/program.cs)]
```
Comment (ahsonkhan) [inline xml/System/Span`1+Enumerator.xml]: Where can I see the code samples?

Comment (mairaw) [reply xml/System/Span`1+Enumerator.xml]: https://github.com/dotnet/samples/pull/85

File: xml/System/Span`1.xml (hunk 1 above)
Comment (rpetrusha) [reply xml/System/Span`1.xml]: How about "without any loss of the source data"?

**Comment by BillWagner** [comment]
Merging based on all reviews.

**Pull request merged**